src/components/fields/Radio.js

```diff
@@ -1,61 +1,34 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import Field from '../Field';
+import * as PropTypes from 'prop-types';
+import { RadioGroupContext } from '../FormContext';
 
-class Radio extends Component {
+export default class Radio extends Component {
   static propTypes = {
-    getValue: PropTypes.func.isRequired,
-    setValue: PropTypes.func.isRequired,
-    setTouched: PropTypes.func.isRequired,
-    onBlur: PropTypes.func,
-    onChange: PropTypes.func,
+    id: PropTypes.string.isRequired,
     value: PropTypes.oneOfType([
       PropTypes.string,
       PropTypes.number,
     ]).isRequired,
   };
 
-  static defaultProps = {
-    type: 'text',
-  };
-
   render() {
-    const {
-      getValue,
-      setValue,
-      setTouched,
-      onBlur,
-      onChange,
-      value,
-      ...props
-    } = this.props;
-
+    const {id, value, ...props} = this.props;
     return (
-      <input
-        {...props}
-        value={value}
-        checked={value == getValue()}
-        type="radio"
-        onChange={event => {
-          if (!event.target.checked) {
-            return;
-          }
-          setValue(event.target.value, () => {
-            if (onChange) {
-              onChange(event);
-            }
-          });
-        }}
-        onBlur={event => {
-          setTouched(() => {
-            if (onBlur) {
-              onBlur(event);
-            }
-          });
-        }}
-      />
+        <RadioGroupContext.Consumer>
+          {({name, chosenItem, getValue}) => {
+            return (
+                <input
+                    {...props}
+                    id={id}
+                    name={name}
+                    type="radio"
+                    onChange={event => chosenItem(event.target.value)}
+                    value={value}
+                    checked={Number.isInteger(value) ? value === +getValue(name) : value === getValue(name)}
```

Comment:
I have doubts about this approach.

Reply:
why? 
```
    static propTypes = {
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    };
```
We expect to receive one of type (string or number only) so I used strict comparison. Better `value == getValue(name)` ?

Reply:
I think better to leave `value == getValue(name)`, also we should fix bug with `null` value using `.hasAttribute('value')`. Take a look at Select component.

Reply:
Radio button with value `null` will also send `true` when we check it using `.hasAttribute('value')`, it will have empty value. We cannot use condition `value == getValue(name)` in attribute `checked`, it will work incorrect.
I can offer usage of checkbox, which will disable radiogroup component at all.